### DISCUSSION TITLE
Parse string based staleTime values

### DISCUSSION THREAD

**Discussion by garand** [TanStack/query#1409] [Ideas] [answered]
I often have to go to Google or some other service to quickly convert a duration (7 days, 1 hr, etc) to milliseconds for use with `staleTime` or the other duration based options for a query. Would you consider adding the option of supplying a string based value to those options?

```js
// Proposed
staleTime: '7d'
refetchInterval: '1hr'
```

With a quick search I came across a library called [Duration.js](https://github.com/icholy/Duration.js) that appears to have the functionality I'm proposing. Of course including extra dependencies, especially only the 2nd one for the whole project isn't awesome. Maybe just build it in natively to the library as a very simplified version.

**Comment by tannerlinsley** [comment, marked as answer]
I would much rather just find and vet a good utility to do this and merely suggest and document its potential usage.

**Reply by garand** [reply]
Probably the right choice.
